src/sentry/integrations/gitlab/integration.py

```diff
@@ -0,0 +1,215 @@
+        ),
+    )
+    name = forms.CharField(
+        label="Gitlab App Name",
+        help_text=_('The GitHub App name of your Sentry app. '
+                    'This can be found on the apps configuration '
+                    'page.'),
+        widget=forms.TextInput(
+            attrs={'placeholder': _('our-sentry-app')}
+        )
+    )
+    group = forms.CharField(
+        label="Gitlab Group Name",
+        widget=forms.TextInput(
+            attrs={'placeholder': _('my-awesome-group')}
+        )
+    )
+    client_id = forms.CharField(
+        label="Gitlab OAuth Client ID",
+        widget=forms.TextInput(
+            attrs={'placeholder': _('1')}
+        )
+    )
+    client_secret = forms.CharField(
+        label="Gitlab OAuth Client Secret",
+        widget=forms.TextInput(
+            attrs={'placeholder': _('XXXXXXXXXXXXXXXXXXXXXXXXXXX')}
+        )
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(InstallationForm, self).__init__(*args, **kwargs)
+
+
+class InstallationConfigView(PipelineView):
+    def dispatch(self, request, pipeline):
+        form = InstallationForm(request.POST)
+        if form.is_valid():
+            form_data = form.cleaned_data
+            form_data['url'] = urlparse(form_data['url']).netloc
+
+            pipeline.bind_state('installation_data', form_data)
+
+            pipeline.bind_state('oauth_config_information', {
+                "access_token_url": u"https://{}/oauth/token".format(form_data.get('url')),
+                "authorize_url": u"https://{}/oauth/authorize".format(form_data.get('url')),
+                "client_id": form_data.get('client_id'),
+                "client_secret": form_data.get('client_secret'),
+            })
+
+            return pipeline.next_step()
+
+        project_form = InstallationForm()
+
+        return render_to_response(
+            template='sentry/integrations/gitlab-config.html',
+            context={
+                'form': project_form,
+            },
+            request=request,
+        )
+
+
+class GitlabIntegrationProvider(IntegrationProvider):
+    key = 'gitlab'
+    name = 'Gitlab'
+    metadata = metadata
+    integration_cls = GitlabIntegration
+
+    needs_default_identity = True
+
+    features = frozenset([
+        IntegrationFeatures.COMMITS,
+        IntegrationFeatures.ISSUE_BASIC,
+    ])
+
+    setup_dialog_config = {
+        'width': 1030,
+        'height': 1000,
+    }
+
+    def _make_identity_pipeline_view(self):
+        """
+        Make the nested identity provider view. It is important that this view is
+        not constructed until we reach this step and the
+        ``oauth_config_information`` is available in the pipeline state. This
+        method should be late bound into the pipeline vies.
+        """
+        identity_pipeline_config = dict(
+            oauth_scopes=(),
+            redirect_url=absolute_uri('/extensions/gitlab/setup/'),
+            verify_ssl=False,
```

Comment:
Why don't we want to verify ssl?

Reply:
similar to GHE, for testing purposes. But should remember to remove this later 😬

Reply:
@markstory actually added in a checkbox for the install form - will put up diff PR for GHE too